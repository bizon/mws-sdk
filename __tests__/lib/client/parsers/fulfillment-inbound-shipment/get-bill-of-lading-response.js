const parseXml = require('../../../../../lib/client/parsers')
const parseGetBillOfLadingResponse = require('../../../../../lib/client/parsers/fulfillment-inbound-shipment/get-bill-of-lading-response')

describe('lib.client.parsers.fulfillment-inbound-shipment.get-bill-of-lading-response', () => {
  it('should parse the GetBillOfLadingResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetBillOfLadingResponse xmlns="http://mws.amazonaws.com/FulfillmentInboundShipment/2010-10-01/">
        <GetBillOfLadingResult>
          <TransportDocument>
            <PdfDocument>ABEAAAAAAlbHMucGRmUEsFBgAAAAABAAEAPwAAALQXAQAAAA==
            </PdfDocument>
            <Checksum>WGQwqA+NlzMVL1plHc/7ZA==</Checksum>
          </TransportDocument>
        </GetBillOfLadingResult>
        <ResponseMetadata>
          <RequestId>985a3fa9-3ce2-46fb-a1c7-321439269d2b</RequestId>
        </ResponseMetadata>
      </GetBillOfLadingResponse>`,
    )

    const res = parseGetBillOfLadingResponse(
      '/fulfillmentInboundShipment:GetBillOfLadingResponse',
      doc,
    )

    expect(res).toMatchSnapshot()
  })
})
