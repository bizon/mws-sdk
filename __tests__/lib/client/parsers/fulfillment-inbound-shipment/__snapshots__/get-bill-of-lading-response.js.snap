// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.fulfillment-inbound-shipment.get-bill-of-lading-response should parse the GetBillOfLadingResponse example response from MWS doc 1`] = `
{
  "getBillOfLadingResult": {
    "transportDocument": {
      "checksum": "WGQwqA+NlzMVL1plHc/7ZA==",
      "pdfDocument": "ABEAAAAAAlbHMucGRmUEsFBgAAAAABAAEAPwAAALQXAQAAAA==",
    },
  },
  "responseMetadata": {
    "requestId": "985a3fa9-3ce2-46fb-a1c7-321439269d2b",
  },
}
`;
